### DISCUSSION TITLE
Feature Request: Bulk Add Items to Collections

### DISCUSSION THREAD

**Discussion by WoutAndAbout** [dani-garcia/vaultwarden#5062] [Ideas]
Hi Vaultwarden team, 

First of all, thank you for developing and maintaining such a fantastic tool! 
I really appreciate all the effort you put into it.

I wanted to ask if it would be possible to add a feature that allows users to bulk add items such as logins to collections. 

Additionally, it would be very helpful to have the option to shift-select an entire list of items, making it easier to multiple add items at once. 

These features would greatly improve the efficiency of managing larger amounts of data.

Thank you in advance for considering this!

**Comment by BlackDex** [comment]
This project does not develop the web-vault, so we will not add any features to it.

Also, we currently are stuck on 2024.6.2 because of missing role features. So any feature in a web-vault version newer then we currently have is probably not going to happen right now.

**Reply by WoutAndAbout** [reply]
Hi,

Thank you for the clarification.

If there’s any future update that allows such a feature, I'd be happy to revisit this request. 
In the meantime, thanks again for all the work you’re doing!

**Comment by Felix-Franz** [comment]
Currently the bulk edit feature is implemented in the web vault, but not completely in the API.

While adding multiple items to a collection is functional, removing multiple items leads to following error:
![image](https://github.com/user-attachments/assets/77e5502b-264a-4cf1-89f7-39ca2845fef0)
```json
{
    "error": "",
    "errorModel": {
        "message": "Bulk removing of collections is not yet implemented",
        "object": "error"
    },
    "error_description": "",
    "exceptionMessage": null,
    "exceptionStackTrace": null,
    "innerExceptionMessage": null,
    "message": "Bulk removing of collections is not yet implemented",
    "object": "error",
    "validationErrors": {
        "": [
            "Bulk removing of collections is not yet implemented"
        ]
    }
}
```

Is there a plan to add this feature to a future version of Vaultwarden? @BlackDex 

## Versions
* Vaultwarden Web: 2025.1.1
* Vaultwarden Server/API: 1.33.2
